src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -112,9 +115,8 @@ class ReportActionContextMenu extends React.Component {
             {
                 text: this.props.translate('reportActionContextMenu.editComment'),
                 icon: Pencil,
-                shouldShow: this.props.reportAction.actorEmail === this.props.session.email
-                    && !isReportMessageAttachment(this.getActionText())
-                    && this.props.reportAction.reportActionID,
+                shouldShow: this.canEdit
```

Comment:
I think this also won't work properly because you have:

```
shouldShow: (some function) && (some boolean)
```

For this one, I would just use an arrow function:

```suggestion
                shouldShow: () => (
                    this.canEdit()
                    && !isReportMessageAttachment(this.getActionText()
                ),
```